### DISCUSSION TITLE
KellerZA integration HA vs. SOFAR Solar 3phase HV hybrid invertor

### DISCUSSION THREAD

**Discussion by RomcaKrestic** [kellerza/sunsynk#220] [Ideas]
Hello, I am sending greetings from the Czech Republic.
I was interested in the integration of KellerZA into Home Assistant and I would like to ask if you have tried this integration on the SOFAR Solar 3-phase HV inverter.
Specifically, my friend and I have a Sofar HYD 10KTL-3PH on the wall, each one his own :-) and I am looking in vain for the possibility of connecting to Home Assistant with your integration.
Thanks for any tips on solutions.
        Beautiful sunny day 😁
        Roman

**Comment by kellerza** [comment]
Hi Roman! 
If you find the modbus protocol/register document you should be able to compare it with the register values in definitions3PH.py.
They might be the same, but even if different you could create your own sensors with mysensors (see the docs) and eventually doing aPR.
